azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/VMImage.java

```diff
@@ -21,6 +21,18 @@
 @AutoValue
 public class VMImage {
 
+   public VMImage(String publisher, String offer, String sku, String version, String location) {
```

Comment:
Don't do this,as it is an auto-value anti-pattern. Leave the default constructor not visible, in default scope, and provide a static factory method to instantiate this. Also replace fields by abstract methods. This helper class should be trated just like any other domain class. Apply this also tot he other helper classes that do not properly implement autovalue.